pkg/hubble/filters/ip.go

```diff
@@ -103,3 +103,33 @@ func (f *IPFilter) OnBuildFilter(ctx context.Context, ff *flowpb.FlowFilter) ([]
 
 	return fs, nil
 }
+
+func filterByIPVersion(ipver []flowpb.IPVersion) (FilterFunc, error) {
+	return func(ev *v1.Event) bool {
+		ver := ev.GetFlow().GetIP().GetIpVersion()
```

Comment:
I see this very line in other places in code, but it makes me uneasy. Can `GetIP` ever be nil?

Reply:
Interestingly, I added a check in my code and then removed it since I didn't find such checks in other places. I would like to know if a) I should add a check in my code b) review all places and add checks in all cases c) first try to create a non-IP scenario and test and check the effect.

Reply:
`GetIP` may return `nil` here (so could `GetFlow` by the way). This is generally fine though, as the protobuf generated functions are `nil` safe: https://github.com/cilium/cilium/blob/389ee9a04225c17529b60597ad85c8fc4b23f8be/api/v1/flow/flow.pb.go#L1453-L1458

Reply:
So no change required here. Thanks for clarifying.

Reply:
Getting back on this, I think that `filterByIPVersion` it should always return `false` when the event is not a Flow. This is what I meant by testing the combination of `IPVersion_IP_NOT_USED` and non-flow events.

When a `FlowFilter` with `IPVersion_IP_NOT_USED` is provided, I'm expecting to have flows returned. Here `ev.GetFlow()` will return `nil` for non-flows events and eventually `GetIpVersion` will return `IPVersion_IP_NOT_USED`, unexpectedly matching my filter.

Reply:
Validated this with a testcase and then fixed it as per your suggestions.